### PR TITLE
Creation of the Multisignature section

### PR DESCRIPTION
**Pull request by lugaxker**
This section describes how OP_CHECKMULTISIG and OP_CHECKMULTISIGVERIFY opcodes work. Feel free to edit it, as English is not my native language.

### CODE REVIEW

File: protocol/blockchain/cryptography/multisignature.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+Multisignature scripts set a condition where N public keys are recorded in the script and at least M of those must provide signatures to unlock the funds. This is also known as *M-of-N multisignature scheme*, where N is the total number of keys and M is the threshold of signatures required for validation. The signatures used can either be ECDSA signatures or Schnorr signatures.
+
+## Public multisignature: OP_CHECKMULTISIG(VERIFY)
+
+Multisig schemes can be built with the opcodes `OP_CHECKMULTISIG` and `OP_CHECKMULTISIGVERIFY`, two opcodes of the Bitcoin Cash [scripting language](/protocol/blockchain/script). `OP_CHECKMULTISIGVERIFY` has the same implementation as `OP_CHECKMULTISIG`, except OP_VERIFY is executed afterward.
+
+The opcode `OP_CHECKMULTISIG` can be included in all sorts of scripts. The minimal [locking script](https://reference.cash/protocol/blockchain/transaction/locking-script) using `OP_CHECKMULTISIG` is:
```
Comment (dagurval) [inline protocol/blockchain/cryptography/multisignature.md:11]: This should be a relative path (to make it independent of `reference.cash`, makes it for example easier for @joshmg to pull changes), but we can fix it in a later commit

File: protocol/blockchain/cryptography/multisignature.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+* Once the first signature has been checked against a matching public key, `checkbits` is bit-shifted to the right (`checkbits := checkbits >> 1`).
+* The process is repeated until all signatures have been checked or not enough public keys remain to produce a successful result.
+* All signatures need to match a public key, otherwise `false` is returned.
+* If the final `checkbits` value is non-zero, then `false` is returned. 
+
+Because public keys are not checked again if they fail any signature comparison (in both cases), signatures must be placed in the unlocking script using the same order as their corresponding public keys were placed in the locking script (P2MS) or redeem script (P2SH).
+
+Note that the `checkbits` element is encoded as a byte array of length `floor((N + 7)/8)` (the shortest byte array that can hold N bits) and must have exactly M bits set to ensure that all signatures are checked against public keys.
+
+To know more about the Schnorr mode, see the [official specification](https://github.com/bitcoincashorg/bitcoincash.org/blob/master/spec/2019-11-15-schnorrmultisig.md).
```
Comment (dagurval) [inline protocol/blockchain/cryptography/multisignature.md:55]: I don't think the term "official" anything exists for the Bitcoin Cash protocol, so just "specification" is better IMO. (Also we want to add a copy of it in this repo)

**Review by dagurval — APPROVED** [review]

**Comment by dagurval** [comment]
Thanks again for quality content!

File: protocol/blockchain/cryptography/multisignature.md (hunk 1 above)
Comment (lugaxker) [reply protocol/blockchain/cryptography/multisignature.md:11]: My bad, I forgot to remove the `https://reference.cash` part here. I will change it in a future commit.

File: protocol/blockchain/cryptography/multisignature.md (hunk 2 above)
Comment (lugaxker) [reply protocol/blockchain/cryptography/multisignature.md:55]: Yes I agree.

File: protocol/blockchain/cryptography/multisignature.md (hunk 1 above)
Comment (lugaxker) [reply protocol/blockchain/cryptography/multisignature.md:11]: Oh you have already changed that. Thanks!

**Pull request merged**